File: python/剑指offer/面试题32 - II. 从上到下打印二叉树 II.py
# Definition for a binary tree node.
from typing import List

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        stack = []
        stack.append(root)
        while stack != []:

            if root.left:
                stack.append(root.left)
            if root.right:
                stack.append(root.right)

File: go/剑指offer/面试题32 - II. 从上到下打印二叉树 II.go
package main

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return [][]int{}
	}
	queue := []*TreeNode{root}
	var res [][]int
	for len(queue) > 0 {
		var tmp []int
		l := len(queue)
		for i := 0; i < l; i += 1 {
			node := queue[0]
			tmp = append(tmp, node.Val)
			if node.Left != nil {
				queue = append(queue, node.Left)
			}
			if node.Right != nil {
				queue = append(queue, node.Right)
			}
			queue = queue[1:]
		}
		res = append(res, tmp)
	}
	return res
}
